packages/google_sign_in/google_sign_in_ios/ios/Classes/FLTGoogleSignInPlugin.m

```diff
@@ -105,104 +102,122 @@ - (instancetype)initWithSignIn:(GIDSignIn *)signIn
+  GIDAuthentication *auth = currentUser.authentication;
+  [auth doWithFreshTokens:^void(GIDAuthentication *authentication, NSError *error) {
+    if (error) {
+      completion(nil, getFlutterError(error));
+    } else {
+      completion([FSITokenData makeWithIdToken:authentication.idToken
+                                   accessToken:authentication.accessToken],
+                 nil);
+    }
+  }];
+}
+
+- (void)signOutWithError:(FlutterError *_Nullable *_Nonnull)error;
+{ [self.signIn signOut]; }
+
+- (void)disconnectWithCompletion:(nonnull void (^)(FlutterError *_Nullable))completion {
+  [self.signIn disconnectWithCallback:^(NSError *error) {
+    // TODO(stuartmorgan): This preserves the pre-Pigeon-migration behavior, but it's unclear why
+    // 'error' is being ignored here.
+    completion(nil);
+  }];
+}
+
+- (void)requestScopes:(nonnull NSArray<NSString *> *)scopes
+           completion:(nonnull void (^)(NSNumber *_Nullable, FlutterError *_Nullable))completion {
+  self.requestedScopes = [self.requestedScopes setByAddingObjectsFromArray:scopes];
+  NSSet<NSString *> *requestedScopes = self.requestedScopes;
+
+  @try {
+    [self.signIn addScopes:requestedScopes.allObjects
+        presentingViewController:[self topViewController]
+                        callback:^(GIDGoogleUser *addedScopeUser, NSError *addedScopeError) {
+                          BOOL granted = NO;
+                          FlutterError *error = nil;
+                          if ([addedScopeError.domain isEqualToString:kGIDSignInErrorDomain] &&
+                              addedScopeError.code == kGIDSignInErrorCodeNoCurrentUser) {
+                            error = [FlutterError errorWithCode:@"sign_in_required"
+                                                        message:@"No account to grant scopes."
+                                                        details:nil];
+                          } else if ([addedScopeError.domain
+                                         isEqualToString:kGIDSignInErrorDomain] &&
+                                     addedScopeError.code ==
+                                         kGIDSignInErrorCodeScopesAlreadyGranted) {
+                            // Scopes already granted, report success.
+                            granted = YES;
+                          } else if (addedScopeUser == nil) {
+                            granted = NO;
+                          } else {
+                            NSSet<NSString *> *grantedScopes =
+                                [NSSet setWithArray:addedScopeUser.grantedScopes];
+                            granted = [requestedScopes isSubsetOfSet:grantedScopes];
+                          }
+                          completion(error == nil ? @(granted) : nil, error);
+                        }];
+  } @catch (NSException *e) {
+    completion(nil, [FlutterError errorWithCode:@"request_scopes" message:e.reason details:e.name]);
+  }
```

Comment:
The diffs in this file look more extensive than they actually are because of indentation changes. The code here was made by:
- creating empty stubs for each of the Pigeon protocol methods,
- moving the body of the relevant `if/else if` block from `handleMethodCall:result:` into that method,
- replacing arg dictionary lookups with the corresponding Pigeon argument (removing some now-unnecessary type-checking code), and
- replacing `result` calls with `completion` calls.